tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,755 @@
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PLAY)
+    pipe_control_api_object.player_pause.assert_called_once()
+    pipe_control_api_object.player_prev.assert_called_once()
+    pipe_control_api_object.player_next.assert_called_once()
+    pipe_control_api_object.player_resume.assert_called_once()
+    # switch away
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_INPUT_SOURCE: SOURCE_NAME_DEFAULT},
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_INPUT_SOURCE] == SOURCE_NAME_DEFAULT
+    # test pipe getting queued externally changes source
+    get_request_return_values["queue"] = SAMPLE_QUEUE_PIPE
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+    await updater_update(["queue"])
+    await hass.async_block_till_done()
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_INPUT_SOURCE] == "librespot-java (pipe)"
+
+
+async def test_librespot_java_play_media(hass, pipe_control_api_object):
+    """Test play media with librespot-java pipe."""
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == initial_state.state
+    assert state.last_updated > initial_state.last_updated
+
+
+async def test_librespot_java_play_media_pause_timeout(hass, pipe_control_api_object):
+    """Test play media with librespot-java pipe."""
+    # test media play with pause timeout
+    pipe_control_api_object.player_pause.side_effect = None
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.CALLBACK_TIMEOUT", 0
+    ):
+        initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        await _service_call(
+            hass,
+            TEST_MASTER_ENTITY_NAME,
+            SERVICE_PLAY_MEDIA,
+            {
+                ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+                ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+            },
+        )
+        state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        assert state.state == initial_state.state
+        assert state.last_updated > initial_state.last_updated
+
+
+async def test_unsupported_update(hass, mock_api_object):
+    """Test unsupported update type."""
+    last_updated = hass.states.get(TEST_MASTER_ENTITY_NAME).last_updated
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+    await updater_update(["config"])
+    await hass.async_block_till_done()
+    assert hass.states.get(TEST_MASTER_ENTITY_NAME).last_updated == last_updated
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)